### ISSUE TITLE
Auto-generated (or not) API Docs

### ISSUE THREAD

**Issue by sfrisk** [jquery/css-chassis#31]
Based on discussions #11, moving the discussion over here to avoid cluttering the Pattern Library issue.

Should we have the API documentation generated via comments in the code, or should it be separate.  This is partly dependent on how we what the final output of the project to be and how easy it is to include generated documentation with that.  Others have also voiced concern in the meeting that discussed this over how this could potentially clutter the code up.

Some generators that have been brought up for possible discussion include:
* Sassdoc
* Topdoc
* hologram
* StyleDocco

**Comment by jzaefferer** [comment]
jQuery Core went through many iterations of documentation formats, starting out with inline/source documentation. We eventually ended up with files in separate GitHub repositories. The workflow to get those updated along with code changes definitely take more effort to manage, but for UI they're definitely worth the gains. For example, in JavaScript, source code rarely can be matched 1:1 to components in the API documentation. I suspect that would apply to [S]CSS as well. I can provide more details on the current state of that setup and its history if there's interest.

**Comment by arschmitz** [comment]
@jzaefferer completely agree for the record im not a fan of documentation generated from source code.

**Comment by trailofdad** [comment]
I've done some research on the topic of auto-generated documentation. There is evidence that auto generated documentation, although convenient, makes for confusing documentation, is repetitious, and clutters code. Here is an academic study that has some good comments from participants: 

http://www.cse.nd.edu/~cmc/papers/mcburney_icpc_2014.pdf 


TL;DR

• “The description is too verbose and contains too many
details.”
• “The summary contains too much information and confuses
the purpose of the method...”
• “The summary seems accurate but too verbose.”
• “Too much information, I cannot understand the comment.”

**Comment by trailofdad** [comment]
With that said, keeping the pattern library and documentation up to date with the source code will have to be built into the workflow to avoid the nightmare of documentation falling behind the code.

**Comment by Florian-R** [comment]
FWIW, there's an [ongoing discussion in WSK](https://github.com/google/web-starter-kit/issues/446) on a similar topic.
